cpp/include/cudf/detail/copy_if.cuh

```diff
@@ -131,7 +131,7 @@ __global__ void scatter_kernel(cudf::mutable_column_device_view output_view,
       temp_data[local_index] = input_view.data<T>()[tid]; // scatter data to shared
 
       // scatter validity mask to shared memory
-      if (input_view.is_valid(tid)) {
+      if (has_validity and input_view.is_valid(tid)) {
```

Comment:
The validity check was being made even though there was no null mask, which was causing internal cuda issue.